### PR TITLE
fix(logs): Cannot set log `removalPolicy: destroy` to more than one LogRetention resources

### PR DESCRIPTION
**Pull request by tmokmss**
Currently the IAM policy for LogRetention custom resource Lambda function is set only when it is initialized. Because that lambda function is a singleton function, it is only initialized once and therefore the IAM policy to remove log groups is not configured properly.

e.g. given we create two LogRetention resources with `removalPolicy: destroy`, the resulting IAM policy has only statement for log group `group1`.

```ts
    new LogRetention(stack, 'MyLambda1', {
      logGroupName: 'group1',
      retention: RetentionDays.ONE_DAY,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    new LogRetention(stack, 'MyLambda2', {
      logGroupName: 'group2',
      retention: RetentionDays.ONE_DAY,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });
```

Also I removed `logs:DeleteLogStream` allow statement because I confirmed it is not required to remove a log group.

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-logs/test/log-retention.test.ts [hunk 1]
```diff
@@ -200,6 +176,137 @@ describe('log retention', () => {
     });
   });
 
+  test('multiple log retention resources with removalPolicy DESTROY', () => {
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-logs/test/log-retention.test.ts:179]: can we break this up into two tests in the following format:

```ts
describe('multiple log retention resources', () => {
  test('both removalPolicy DESTROY', () => {
  });
  test('with removalPolicy DESTROY and removalPolicy RETAIN', () => {
  });
});
```

hopefully that make sense. just trying to split these up into units and have descriptive test names.

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @tmokmss! I have a few minor nits but otherwise this looks good.

File: packages/@aws-cdk/aws-logs/test/log-retention.test.ts [hunk 2]
```diff
@@ -200,6 +176,137 @@ describe('log retention', () => {
...
+    });
+
+    new LogRetention(stack, 'MyLambda3', {
+      logGroupName: 'group3',
+      retention: RetentionDays.ONE_DAY,
+      removalPolicy: cdk.RemovalPolicy.RETAIN,
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
```
[the remark this answers is not in the document]
Comment (tmokmss) [reply packages/@aws-cdk/aws-logs/test/log-retention.test.ts]: Yes, previously this would result in a statement only for log group1.

**Comment by tmokmss** [comment]
Hi @kaizencc, thank you for the review! I addressed all of your comments :)

File: packages/@aws-cdk/aws-logs/lib/log-retention.ts [hunk 3]
```diff
@@ -210,4 +193,20 @@ class LogRetentionFunction extends Construct implements cdk.ITaggable {
       }
     });
   }
+
+  /**
+   * @internal
+   */
+  grantDeleteLogGroup(logGroupName: string) {
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-logs/lib/log-retention.ts:200]:
```suggestion
  public grantDeleteLogGroup(logGroupName: string) {
```

**Review by kaizencc — APPROVED** [review]
Thanks @tmokmss!

**Review by kaizencc — APPROVED** [review]

**Pull request merged**